### DISCUSSION TITLE
Count Only Code Changed in The Current Branch

### DISCUSSION THREAD

**Discussion by DarwinJS** [boyter/scc#280] [General]
@boyter - is there way to only count the *changed* code in the current branch?

I am especially interested in whether the complexity measures would still be relevant if it can be setup this way. Looking for a way to measure the complexity of changes in a branch / pr.

**Comment by DarwinJS** [comment]
I guess I can pipe `git diff --name-only` into scc to get a complete file complexity and then also show the number of changes with `git diff --numstat`.

Still interested in a way to get only the changes if it is possible :)

**Reply by DarwinJS** [reply]
Getting closer - is anyone an awk genius and can help me get "git changes" as a column on the output?

`scc $(git diff --name-only $(git merge-base HEAD master) -- HEAD) --by-file -s complexity`

**Comment by boyter** [comment]
So nothing native in there. It is something I have considered doing... as I actually wrote a plugin for bitbucket which did something similar. Native support would mean having some level of git integration though which I am not crazy about.

You seem to be going down the right path with some sort of awk shell goodness. If you come up with one let me know and ill add it to the README as I am sure other people would like it too.

**Comment by DarwinJS** [comment]
This is complete and available at [Git Diff Revision Activity Metrics](https://gitlab.com/guided-explorations/ci-cd-plugin-extensions/git-diff-revision-activity-metrics)

**Reply by DarwinJS** [reply]
![diff-factor-and-level-in-pipeline-vars](https://user-images.githubusercontent.com/11597190/128886813-cde3be1b-5c71-44b4-919e-150c3a925383.png)
![easy-mr-access-with-expose-as](https://user-images.githubusercontent.com/11597190/128886815-d4389db2-34d3-4a19-bab5-1d559b7e8b27.png)
![mr-diff-report](https://user-images.githubusercontent.com/11597190/128886816-96ca8e19-6284-4728-8f8f-9a06c7f53519.png)
![mr-list-with-labels](https://user-images.githubusercontent.com/11597190/128886817-22951f0c-d470-4f76-b4fd-baee8bc9d4eb.png)
![mr-report-in-ci-log](https://user-images.githubusercontent.com/11597190/128886818-5c1dc2d8-d442-4ffc-9323-fbc8561d4139.png)
